tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/decimaltype/BDecimalValueNegativeTest.java

```diff
@@ -33,15 +33,13 @@
     @Test
     public void testDecimalValue() {
         CompileResult compileResult = BCompileUtil.compile("test-src/types/decimal/decimal_value_negative.bal");
-        Assert.assertEquals(compileResult.getErrorCount(), 19);
+        Assert.assertEquals(compileResult.getErrorCount(), 18);
         int index = 0;
         BAssertUtil.validateError(compileResult, index++, "missing semicolon token", 21, 1);
         BAssertUtil.validateError(compileResult, index++, "missing binary operator", 23, 21);
         BAssertUtil.validateError(compileResult, index++, "undefined symbol 'g'", 23, 21);
         BAssertUtil.validateError(compileResult, index++, "leading zeros in numeric literals", 26, 17);
         BAssertUtil.validateError(compileResult, index++, "Hexadecimal '-0x' too small", 29, 17);
-        BAssertUtil.validateError(compileResult, index++, "incompatible types: expected 'decimal', found 'int'", 29,
```

Comment:
This is for `decimal e = -0xX1231.12P1;`. 

https://ballerina.io/ballerina-spec/spec.html#section_5.2.4.2 says 

> Any decimal value can be represented by a DecimalFloatingPointNumber with an optional DecimalTypeSuffix.

So is it correct for this error to no longer be logged?

Reply:
https://ballerina.io/ballerina-spec/spec.html#section_6.5 also mentions this (before we consider the expected type).

> if the numeric-literal is a HexFloatingPointLiteral, then the basic type is float;